src/main/java/net/imglib2/algorithm/stats/InformationMetrics.java

```diff
@@ -0,0 +1,218 @@
+import net.imglib2.view.Views;
+
+/**
+ * This class provides method for computing information metrics (entropy, mutual information)
+ * for imglib2.
+ * 
+ * @author John Bogovic
+ */
+public class InformationMetrics
+{
+
+	/**
+	 * Returns the normalized mutual information of the inputs
+	 * @param rai the RandomAccessibleInterval
+	 * @param ra the RandomAccessible
+	 * @return the normalized mutual information
+	 */
+	public static <T extends RealType< T >> double normalizedMutualInformation(
+			final IterableInterval< T > dataA,
+			final IterableInterval< T > dataB,
+			double histmin, double histmax, int numBins )
+	{
+		HistogramNd< T > jointHist = jointHistogram( dataA, dataB, histmin, histmax, numBins );
+		return normalizedMutualInformation( jointHist );
+	}
+
+	/**
+	 * Returns the normalized mutual information of the inputs
+	 * @param jointHist the joint histogram
+	 * @return the normalized mutual information
+	 */
+	public static <T extends RealType< T >> double normalizedMutualInformation(
+		final HistogramNd< T > jointHist )
+	{
+		double HA = marginalEntropy( jointHist, 0 );
+		double HB = marginalEntropy( jointHist, 1 );
+		double HAB = entropy( jointHist );
+		return 2 * ( HA + HB - HAB ) / ( HA + HB );
+	}
+	
+	/**
+	 * Returns the normalized mutual information of the inputs
+	 * @param rai the RandomAccessibleInterval
+	 * @param ra the RandomAccessible
+	 * @return the normalized mutual information
+	 */
+	public static <T extends RealType< T >> double mutualInformation(
+			IterableInterval< T > dataA,
+			IterableInterval< T > dataB,
+			double histmin, double histmax, int numBins )
+	{
+		HistogramNd< T > jointHist = jointHistogram( dataA, dataB, histmin, histmax, numBins );
+
+		double HA = marginalEntropy( jointHist, 0 );
+		double HB = marginalEntropy( jointHist, 1 );
+		double HAB = entropy( jointHist );
+
+		return HA + HB - HAB;
+	}
+
+	/**
+	 * Returns the normalized mutual information of the inputs
+	 * @param ra the RandomAccessible
+	 * @return the normalized mutual information
+	 */
+	public static <T extends RealType< T >> double mutualInformation(
+			final HistogramNd< T > jointHist )
+	{
+		double HA = marginalEntropy( jointHist, 0 );
+		double HB = marginalEntropy( jointHist, 1 );
+		double HAB = entropy( jointHist );
+		return HA + HB - HAB;
+	}
+	
+	/**
+	 * Compute a the joint histogram.
+	 * @param dataA
+	 * @param dataB
+	 * @param histmin
+	 * @param histmax
+	 * @param numBins
+	 * @return the joint histogram
+	 */
+	public static < T extends RealType< T > > HistogramNd< T > jointHistogram( 
+			final IterableInterval< T > dataA,
+			final IterableInterval< T > dataB,
+			final double histmin,
+			final double histmax,
+			final int numBins )
+	{
+		Real1dBinMapper< T > binMapper = new Real1dBinMapper< T >( histmin, histmax, numBins, false );
+		ArrayList< BinMapper1d< T > > binMappers = new ArrayList< BinMapper1d< T > >( 2 );
+		binMappers.add( binMapper );
+		binMappers.add( binMapper );
```

Comment:
Not tested but something like this should be possible:
```suggestion
		List< BinMapper1d< T > > binMappers = Arrays.asList( binMapper, binMapper );
```